src/frontend/src/utils/with_options.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::num::NonZeroU32;
+
+use itertools::Itertools;
+use risingwave_common::error::{ErrorCode, RwError};
+use risingwave_sqlparser::ast::{
+    CreateSinkStatement, CreateSourceStatement, SqlOption, Statement, Value,
+};
+
+use crate::catalog::source_catalog::KAFKA_CONNECTOR;
+
+mod options {
+    use risingwave_common::catalog::hummock::PROPERTIES_RETENTION_SECOND_KEY;
+
+    pub const APPEND_ONLY: &str = "appendonly";
+    pub const CONNECTOR: &str = "connector";
+    pub const RETENTION_SECONDS: &str = PROPERTIES_RETENTION_SECOND_KEY;
+}
+
+/// Options or properties extracted from the `WITH` clause of DDLs.
+#[derive(Default, Clone, Debug, PartialEq)]
+pub struct WithOptions {
```

Comment:
I'm not sure where to put this and how to name it. Postgres names the clauses after `WITH` "option"s, so I follow the convention while keeping some "properties" naming in our system untouched.